src/components/web3/api.ts

```diff
@@ -0,0 +1,309 @@
+    }
+
+    const page = await response.json();
+    let { nfts } = page;
+    const { next } = page;
+    if (next) {
+      nfts = nfts.concat(await getNFTs(next));
+    }
+
+    const collections = nfts.reduce(
+      (collections: { [key: string]: NFTcollection }, nft: NFT) => {
+        if (
+          nft?.collection?.collection_id &&
+          !collections[nft.collection.collection_id]
+        ) {
+          collections[nft.collection.collection_id] = {
+            id: nft.collection.collection_id,
+            name: nft.collection.name,
+            image_url: nft.image_url,
+          };
+        }
+
+        return collections;
+      },
+      {}
+    );
+
+    return Object.values(collections);
+  };
+
+  try {
+    const getNFTsByWalletURL = `${SIMPLEHASH_PROXY_ROOT_URL}/api/v0/nfts/owners?chains=ethereum&wallet_addresses=${address}`;
+    console.log(`>>> GET ${getNFTsByWalletURL}`);
+    return getNFTs(getNFTsByWalletURL);
+  } catch (error: any) {
+    console.error("!!! web3NFTcollections", error);
+    throw error;
+  }
+};
+
+const getNonce = async (): Promise<string> => {
+  const getNonceURL = "/api/v1/nonce";
+  const response = await fetchWithTimeout(getNonceURL, {
+    method: "GET",
+    headers: {
+      accept: "application/json",
+    },
+  });
+
+  const { nonce } = await response.json();
+  return nonce;
+};
+
+const STORAGE_KEY = "web3.authentication";
+
+export const web3Prove = async (
+  web3Address: string
+): Promise<Web3Authentication> => {
+  if (!web3Address) {
+    throw new Error("not logged into Web3");
+  }
+
+  window.web3 = new ethers.BrowserProvider(window.ethereum);
+  if (!window.web3) {
+    throw new Error("unable to create ethers.BrowserProvider object");
+  }
+
+  const nonce = await getNonce();
+  console.log("!!! nonce", nonce);
+  const message: EIP4361Message = {
+    domain: "talk.brave.com",
+    address: web3Address,
+    statement:
+      "Please sign this message so Brave Talk knows that you own this address",
+    uri: "https://talk.brave.com",
+    version: "1",
+    chainId: 1,
+    // HT:https://stackoverflow.com/questions/40031688/javascript-arraybuffer-to-hex/40031979
+    // btoa has some characters not allowed by the EIP-4361 ABNF
+    nonce: nonce, //[...nonce].map((x) => x.toString(16).padStart(2, "0")).join(""),
+    issuedAt: new Date().toISOString(),
+  };
+  const payload = createEIP4361Message(message);
+  const payloadBytes = new TextEncoder().encode(payload);
+  const hexPayload = hexlify(payloadBytes);
+  const signer = await window.web3.getSigner(web3Address);
+  const signature = await signer.signMessage(payload);
+  console.log(`!!! web3 signature=${signature}`);
+
+  const result = {
+    method: "EIP-4361-json",
+    proof: {
+      signer: web3Address,
+      signature: signature,
+      payload: hexPayload,
+    },
+  };
+
+  window.localStorage.setItem(STORAGE_KEY, JSON.stringify(result));
```

Comment:
What's the intent of storing this locally? Once it's been submitted we won't be able to reuse it because of the nonce replay protection we added. I think we can eliminate the `web3RestoreAuth` function all together too if we're not going to be storing this in localStorage.